components/camel-sip/src/main/java/org/apache/camel/component/sip/SipConfiguration.java

```diff
@@ -243,32 +483,40 @@ public void parseURI() throws Exception {
         LOG.trace("Consumer:" + consumer + " StackName:" + stackName);
         LOG.trace("From User: " + getFromUser() + " From host: " + getFromHost() + " From Port: " + getFromPort());
          
-        createFactoriesAndHeaders(parameters, component);
-        
+        this.createFactoriesAndHeaders(parameters, component);
+
         sipUri = component.resolveAndRemoveReferenceParameter(parameters, "sipUri", SipURI.class, null);
-        if (sipUri == null) {
+        if (sipUri == null && (!consumer  || subscribing)) {
             sipUri = addressFactory.createSipURI(getToUser(), getToHost() + ":" + getToPort());
         }
 
+        //throws exceptions when these mandatory instances end up as null
         ObjectHelper.notNull(fromUser, "From User");
         ObjectHelper.notNull(fromHost, "From Host");
         ObjectHelper.notNull(fromPort, "From Port");
         ObjectHelper.notNull(eventHeader, "Event Header");
         ObjectHelper.notNull(eventHeaderName, "Event Header Name");        
         ObjectHelper.notNull(eventId, "Event Id");        
-    }    
+    }
 
+    /**
+     * Creates the header, address and message factories along with the all the header instances.
+     *
+     * @param parameters the parameters given when endpoint got created
+     * @param component the SipComponent
+     * @throws Exception when something goes wrong looking up from the parameters map
+     */
     @SuppressWarnings("unchecked")
     private void createFactoriesAndHeaders(Map<String, Object> parameters, SipComponent component) throws Exception {
         headerFactory = sipFactory.createHeaderFactory();
         addressFactory = sipFactory.createAddressFactory();
-        setMessageFactory(sipFactory.createMessageFactory());
+        this.setMessageFactory(sipFactory.createMessageFactory());
         
         fromHeader = component.resolveAndRemoveReferenceParameter(parameters, "fromHeader", FromHeader.class, null);
         if (fromHeader == null) { 
             createFromHeader();
         }
-        if (!presenceAgent) {
+        if(!consumer || subscribing) {
```

Comment:
@nikvaessen you use this expression in several places, could you write a helper function that returns a boolean value to encapsulate this expression?